docs/src/internals/runtime.md

```diff
@@ -4,18 +4,18 @@ In the previous sections we focused on building applications in Crux and using
 its public APIs to do so. In this and the following chapters, we'll look at how
 the internals of Crux work, starting with the capability runtime.
 
-The capability runtime is a set of components whose job it is to facilitate the
-processing of effects to present the two perspectives we previously mentioned:
+The capability runtime is a set of components that processes effects, presenting
+the two perspectives we previously mentioned:
 
 - For the core, the shell appears to be a platform with a message based system
   interface
 - For the shell, the core appears as a stateful library responding to events
   with request for side-effects
 
 There are a few challenges to solve in order to facilitate this interface.
-First, each run of the `update` function can call several capabilities, and the
-requested effects are expected to be emitted together and processed
-concurrently, so the calls can't be blocking. Second, each effect requested from
+First, each run of the `update` function can call several capabilities. The
+requested effects are expected to be emitted together, and each group of effects
+will be processed concurrently, so the calls can't be blocking. Second, each effect requested from
```

Comment:
I'm trying to capture here the fact that mutliple effects emitted from a single run of `update` happen sequentially, but each "batch" or "group" of effects is processed concurrently. Maybe "batch" is a better term.